source/android/adaptivecards/src/main/cpp/objectmodel_wrap.cpp

```diff
@@ -26135,6 +26251,42 @@ SWIGEXPORT jlong JNICALL Java_io_adaptivecards_objectmodel_AdaptiveCardObjectMod
 }
 
 
+SWIGEXPORT jlong JNICALL Java_io_adaptivecards_objectmodel_AdaptiveCardObjectModelJNI_UnknownActionParser_1DeserializeFromString(JNIEnv *jenv, jclass jcls, jlong jarg1, jobject jarg1_, jlong jarg2, jobject jarg2_, jstring jarg3) {
+  jlong jresult = 0 ;
+  AdaptiveCards::UnknownActionParser *arg1 = (AdaptiveCards::UnknownActionParser *) 0 ;
+  AdaptiveCards::ParseContext *arg2 = 0 ;
+  std::string *arg3 = 0 ;
+  std::shared_ptr< AdaptiveCards::UnknownActionParser > *smartarg1 = 0 ;
+  std::shared_ptr< AdaptiveCards::BaseActionElement > result;
+  
+  (void)jenv;
+  (void)jcls;
+  (void)jarg1_;
+  (void)jarg2_;
+  
+  smartarg1 = *(std::shared_ptr<  AdaptiveCards::UnknownActionParser > **)&jarg1;
+  arg1 = (AdaptiveCards::UnknownActionParser *)(smartarg1 ? smartarg1->get() : 0); 
+  
+  arg2 = (AdaptiveCards::ParseContext *)((*(std::shared_ptr<  AdaptiveCards::ParseContext > **)&jarg2) ? (*(std::shared_ptr<  AdaptiveCards::ParseContext > **)&jarg2)->get() : 0);
+  if (!arg2) {
+    SWIG_JavaThrowException(jenv, SWIG_JavaNullPointerException, "AdaptiveCards::ParseContext & reference is null");
+    return 0;
+  } 
+  if(!jarg3) {
+    SWIG_JavaThrowException(jenv, SWIG_JavaNullPointerException, "null string");
```

Comment:
>null string [](start = 66, length = 11)

our exception messages could do with some work? perhaps more specific (not sure what arg3 is here ) but perhaps include a bit more context on what "string" alludes to?

Reply:
This is glue code generated by an automated tool called [SWIG](http://www.swig.org/) :)

---
In reply to: [271977090](https://github.com/Microsoft/AdaptiveCards/pull/2612#discussion_r271977090) [](ancestors = 271977090)

Reply:
general rule -- things in `source/android/adaptivecards/src/main/*.[h|cpp]` and everything under `source/android/adaptivecards/main/java/io/adaptivecards/objectmodel` are generated files

---
In reply to: [271978478](https://github.com/Microsoft/AdaptiveCards/pull/2612#discussion_r271978478) [](ancestors = 271978478,271977090)